src/element/mod.rs

```diff
@@ -663,66 +670,82 @@ impl Element {
         }
     }
 
-    /// Serializes this [`Element`] as binary Ion, returning the output as a `Vec<u8>`.
-    ///
-    /// This is a convenience method; it is less efficient than [`Element::write_to`] because:
-    /// 1. It must allocate a new `Vec<u8>` to fill and return.
-    /// 2. It encodes this [`Element`] as its own binary Ion stream, limiting the benefit of the
-    ///    symbol table.
-    ///
-    /// ```
-    ///# use ion_rs::IonResult;
-    ///# fn main() -> IonResult<()> {
-    /// use ion_rs::element::Element;
-    /// use ion_rs::ion_list;
-    ///
-    /// // Construct an Element
-    /// let element_before: Element = ion_list! [1, 2, 3].into();
-    ///
-    /// // Write the Element to a buffer as binary Ion
-    /// let binary_ion: Vec<u8> = element_before.to_binary()?;
-    ///
-    /// // Read the Element back from the serialized form
-    /// let element_after = Element::read_one(&binary_ion)?;
-    ///
-    /// // Confirm that no data was lost
-    /// assert_eq!(element_before, element_after);
-    ///# Ok(())
-    ///# }
-    /// ```
+    #[doc = r##"
+Serializes this [`Element`] as binary Ion, returning the output as a `Vec<u8>`.
+"##]
+    #[cfg_attr(
+        feature = "experimental-writer",
+        doc = r##"
+This is a convenience method; it is less efficient than [`Element::write_to`] because:
+1. It must allocate a new `Vec<u8>` to fill and return.
+2. It encodes this [`Element`] as its own binary Ion stream, limiting the benefit of the
+   symbol table.
+"##
+    )]
+    #[doc = r##"
+```
+# use ion_rs::IonResult;
+# fn main() -> IonResult<()> {
+use ion_rs::element::Element;
+use ion_rs::ion_list;
+
+// Construct an Element
+let element_before: Element = ion_list! [1, 2, 3].into();
+
+// Write the Element to a buffer as binary Ion
+let binary_ion: Vec<u8> = element_before.to_binary()?;
+
+// Read the Element back from the serialized form
+let element_after = Element::read_one(&binary_ion)?;
+
+// Confirm that no data was lost
+assert_eq!(element_before, element_after);
+# Ok(())
+# }
+```
+"##]
     pub fn to_binary(&self) -> IonResult<Vec<u8>> {
         let mut buffer = Vec::new();
         self.write_as(Format::Binary, &mut buffer)?;
         Ok(buffer)
     }
 
-    /// Serializes this [`Element`] as text Ion using the requested [`TextKind`] and returning the
-    /// output as a `String`.
-    ///
-    /// This is a convenience method; to provide your own buffer instead of allocating a `String`
-    /// on each call, see [`Element::write_as`]. To provide your own writer instead of constructing
-    /// a new `String` and writer on each call, see [`Element::write_to`].
```

Comment:
🗺️ This doc comment referred to `write_to`, which is feature gated. If you attempted to build the docs without `"experimental-writer"` enabled, it would fail because `write_to` was not available.

This change hides that section when `write_to` is not available. The content of the comment is otherwise unchanged.